src/SMG/ManagerBundle/Controller/Traits/HandleUserTrait.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace SMG\ManagerBundle\Controller\Traits;
+
+trait HandleUserTrait
+{
+    /**
+     * Check if the information in $user is enough
+     * and valid to create a new User in database.
+     *
+     * @param User $user user to validate
+     *
+     * @return array contains the error(s) list,
+     * empty if no error
+     */
+    private function validates($user)
+    {
+        $validationRules = array('mobile_app_registration');
+
+        if (!is_null($user->getEmail())) {
+            $validationRules[] = 'with_email';
+        }
+
+        if (!is_null($user->getPhoneNumber())) {
+            $validationRules[] = 'with_phone';
+        }
+
+        $validator = $this->container->get('validator');
+        $errors = $validator->validate(
+            $user,
+            $validationRules
+        );
+
+        return $errors;
+    }
+
+    /**
+     *
+     */
+    private function generateToken()
```

Comment:
i think it was used by the function you remove in preivous commit from the trait, so it can be remove from the trait too